src/iam/iam-base.ts

```diff
@@ -67,6 +70,14 @@ export type Transaction = {
   from: string;
 };
 
+export type ChainConfig = {
+  rpcUrl: string,
```

Comment:
If we're requiring the `rpcUrl` in `ChainConfig` (which I think we need for WalletConnect, right?), could we remove it from `ConnectionOptions`? I think it is confusing that it is in both places

Reply:
Unfortunately we need both of them, because the one which is defined in `ConnectionOptions` along with `privateKey` is a way to connect to blockchain instead of providing wallet. `rpcUrl` defined in `ChainConfig` is not input value, but part of the mapping used by walled to resolve url from chain id